wolfcrypt/src/asn.c

```diff
@@ -24547,7 +24587,7 @@ static int MakeCertReq(Cert* cert, byte* derBuffer, word32 derSz,
         {
             /* Encode subject name into space in buffer. */
             ret = SetNameEx((byte*)dataASN[2].data.buffer.data,
-                    dataASN[3].data.buffer.length, &cert->subject, cert->heap);
+                    dataASN[2].data.buffer.length, &cert->subject, cert->heap);
```

Comment:
For reference this was one bug I found and with cert request and ASN template (index should be 2 here).